src/System.CommandLine.Tests/GetValueByNameParserTests.cs

```diff
@@ -135,14 +135,88 @@ public void When_required_argument_value_is_not_parsed_then_an_exception_is_thro
         [Fact]
         public void When_non_existing_name_is_used_then_exception_is_thrown()
         {
-            ParseResult parseResult = new Command("noSymbols").Parse("");
+            const string nonExistingName = "nonExisting";
+            Command command = new ("noSymbols");
+            ParseResult parseResult = command.Parse("");
 
-            Action getRequired = () => parseResult.GetValue<int>("required");
+            Action getRequired = () => parseResult.GetValue<int>(nonExistingName);
 
             getRequired
                 .Should()
                 .Throw<InvalidOperationException>()
-                .Where(ex => ex.Message == LocalizationResources.RequiredArgumentMissing(parseResult.FindResultFor(command.Arguments[0])));
+                .Where(ex => ex.Message == $"No symbol result found for \"{nonExistingName}\" for command \"{command.Name}\".");
+        }
+
+        [Fact]
+        public void When_an_option_and_argument_use_same_name_on_the_same_level_of_the_tree_an_exception_is_thrown()
+        {
+            const string sameName = "same";
+
+            RootCommand command = new()
+            {
+                new Argument<int>(sameName)
+                {
+                    Arity = ArgumentArity.ZeroOrOne
+                },
+                new Option<int>(sameName)
+            };
+
+            ParseResult parseResult = command.Parse("");
+
+            Action getConflicted = () => parseResult.GetValue<int>(sameName);
+
+            getConflicted
+                .Should()
+                .Throw<NotSupportedException>()
+                .Where(ex => ex.Message == $"More than one symbol uses name \"{sameName}\" for command \"{command.Name}\".");
```

Comment:
I feel like we should be detecting this at parser configuration time, e.g. under `ThrowIfInvalid` or earlier.

Reply:
The name is immutable now, so we should be able to throw at the moment when given symbol is being added to the Command.